Assets/Scripts/zombieScript.cs

```diff
@@ -134,12 +122,26 @@ protected override void Move()
 
     protected override void Attack()
     {
-        agent.isStopped = true; // NavMeshAgent.Stop is obsolete. Set NavMeshAgent.isStopped to true.
-        
-        targetEnemy.TakeDamage(damage);
+        if (enumerator == null)
+        {
+            agent.isStopped = true; // NavMeshAgent.Stop is obsolete. Set NavMeshAgent.isStopped to true.
+
+            int randomNumber = Random.Range(0, 2);
 
-        enumerator = ResetAttack();
-        StartCoroutine(enumerator);
+            if (randomNumber == 0)
```

Comment:
change this to a ternary operation (condition ? valueA : valueB) instead of an if else